### ISSUE TITLE
Support numpad input

### ISSUE THREAD

**Issue by falkTX** [lv2/pugl#70]
I am using pugl for a port of a tool that originally uses GLFW.
Compared to GLFW, pugl can do all the same keyboard input but it is missing these definitions:

```
#define GLFW_KEY_KP_0               320
#define GLFW_KEY_KP_1               321
#define GLFW_KEY_KP_2               322
#define GLFW_KEY_KP_3               323
#define GLFW_KEY_KP_4               324
#define GLFW_KEY_KP_5               325
#define GLFW_KEY_KP_6               326
#define GLFW_KEY_KP_7               327
#define GLFW_KEY_KP_8               328
#define GLFW_KEY_KP_9               329
#define GLFW_KEY_KP_DECIMAL         330
#define GLFW_KEY_KP_DIVIDE          331
#define GLFW_KEY_KP_MULTIPLY        332
#define GLFW_KEY_KP_SUBTRACT        333
#define GLFW_KEY_KP_ADD             334
#define GLFW_KEY_KP_ENTER           335
#define GLFW_KEY_KP_EQUAL           336
```

Seems it is all related to numpad stuff. Would be great to have this supported in pugl, I can help with adding support for it but not really sure how to start on that.

**Comment by drobilla** [comment]
I dunno.  It could be done with a bunch of tedious boilerplate, but this bit of the API is really a toy, and numpads are pretty rare these days.

I don't really know what to do about that without having to write down every key symbol that exists on every type of keyboard in the world.  For learned bindings you probably want to use the scancode, but there's currently no way to turn that into a more portable identifier.

Maybe the "keysym" (the thing jammed into the unicode reserved area in pugl's case) doesn't need to be portable at all.  That'd probably make life easier.

**Comment by falkTX** [comment]
> numpads are pretty rare these days

hell no, I dont have a single keyboard without numpad here, I need them!

maybe worth checking how how glfw does it, since it has defines for numpad stuff. should be possible to do it in a portable way, in theory.

**Comment by drobilla** [comment]
Turns out this is a bit annoying at least on X11 because of num lock.  You need XCB or to hardcode the mapping between the arrows and the numbers or... something... it's not just adding some switch cases, anyway.

The other platforms seem straightforward.

**Comment by drobilla** [comment]
aaaaaaaaand I somehow lost the branch where I figured that out :/

**Comment by drobilla** [comment]
Implemented in a347ffd
